pkg/sdk/sdk.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package sdk
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/absmach/supermq/pkg/errors"
+	"moul.io/http2curl"
+)
+
+const (
+	// CTJSON represents JSON content type.
+	CTJSON ContentType = "application/json"
+
+	// CTJSONSenML represents JSON SenML content type.
+	CTJSONSenML ContentType = "application/senml+json"
+
+	// CTBinary represents binary content type.
+	CTBinary ContentType = "application/octet-stream"
+
+	// EnabledStatus represents enable status for a client.
+	EnabledStatus = "enabled"
+
+	// DisabledStatus represents disabled status for a client.
+	DisabledStatus = "disabled"
+
+	BearerPrefix = "Bearer "
+
+	ClientPrefix = "Client "
+)
+
+// ContentType represents all possible content types.
+type ContentType string
+
+var _ SDK = (*mgSDK)(nil)
+
+type PageMetadata struct {
+	Total    uint64   `json:"total"`
+	Offset   uint64   `json:"offset"`
+	Limit    uint64   `json:"limit"`
+	Metadata Metadata `json:"metadata,omitempty"`
+}
+
+// SDK contains SuperMQ API.
+//
+//go:generate mockery --name SDK --output=./mocks --filename sdk.go --quiet --note "Copyright (c) Abstract Machines"
+type SDK interface {
```

Comment:
MG SDK should include SMQ SDK so you can combine MG features and SMQ features using a single tool.

Reply:
I cannot complete this change before the methods are removed from the SDK interface in supermq

Reply:
You mean until they are removed from SMQ SDK due to name conflict?